ntlm3/ntlm.py

```diff
@@ -134,282 +56,252 @@
-        returns
-            24-byte buffer to contain the LM response upon return
-    """
-    # padding with zeros to make the hash 21 bytes long
-    password_hash += b'\0' * (21 - len(password_hash))
-
-    res = b''
-    dobj = des.DES(password_hash[0:7])
-    res = res + dobj.encrypt(server_challenge[0:8])
-
-    dobj = des.DES(password_hash[7:14])
-    res = res + dobj.encrypt(server_challenge[0:8])
-
-    dobj = des.DES(password_hash[14:21])
-    res = res + dobj.encrypt(server_challenge[0:8])
-    return res
-
-
-def ComputeResponse(ResponseKeyNT, ResponseKeyLM, ServerChallenge, ServerName, ClientChallenge=b'\xaa' * 8,
-                    Time=b'\0' * 8):
-    LmChallengeResponse = hmac.new(ResponseKeyLM, ServerChallenge + ClientChallenge).digest() + ClientChallenge
-
-    Responserversion = b'\x01'
-    HiResponserversion = b'\x01'
-    temp = Responserversion + HiResponserversion + b'\0' * 6 + Time + ClientChallenge + b'\0' * 4 + ServerChallenge + b'\0' * 4
-    NTProofStr = hmac.new(ResponseKeyNT, ServerChallenge + temp).digest()
-    NtChallengeResponse = NTProofStr + temp
+    # Getting the values from the CHALLENGE_MESSAGE
+    signature = msg2[0:8]
+    assert signature == NTLM_SIGNATURE
 
-    # TODO - This variable isn't used
-    SessionBaseKey = hmac.new(ResponseKeyNT, NTProofStr).digest()  # noqa
+    message_type = struct.unpack("<I", msg2[8:12])[0]
+    assert message_type == MessageTypes.NTLM_CHALLENGE
 
-    return (NtChallengeResponse, LmChallengeResponse)
+    # TargetName Fields
+    target_name_len = struct.unpack("<H", msg2[12:14])[0]
+    target_name_max_len = struct.unpack("<H", msg2[14:16])[0]
+    target_name_buffer_offset = struct.unpack("<I", msg2[16:20])[0]
 
+    negotiate_flags = struct.unpack("<I", msg2[20:24])[0]
+    server_challenge = msg2[24:32]
+    reserved = msg2[32:40]
 
-def ntlm2sr_calc_resp(ResponseKeyNT, ServerChallenge, ClientChallenge=b'\xaa' * 8):
+    if negotiate_flags & NegotiateFlags.NTLMSSP_REQUEST_TARGET:
+        target_name = msg2[target_name_buffer_offset:target_name_buffer_offset + target_name_len]
 
-    LmChallengeResponse = ClientChallenge + b'\0' * 16
-    sess = hashlib.md5(ServerChallenge + ClientChallenge).digest()
-    NtChallengeResponse = calc_resp(ResponseKeyNT, sess[0:8])
-    return (NtChallengeResponse, LmChallengeResponse)
+    if negotiate_flags & NegotiateFlags.NTLMSSP_NEGOTIATE_TARGET_INFO:
+        # TargetInfo Fields
+        target_info_len = struct.unpack("<H", msg2[40:42])[0]
+        target_info_max_len = struct.unpack("<H", msg2[42:44])[0]
+        target_info_buffer_offset = struct.unpack("<I", msg2[44:48])[0]
 
+        target_info_raw = msg2[target_info_buffer_offset:target_info_buffer_offset + target_info_len]
+        target_info = TargetInfo(target_info_raw)
+    else:
+        target_info = None
 
-def create_LM_hashed_password_v1(passwd):
-    """create LanManager hashed password"""
+    return (server_challenge, negotiate_flags, target_info)

```

Comment:
This will probably cause some issues for people, because of the added support for NTLMv2, we need to return the target_info from the CHALLENGE_MESSAGE. Will require some changes on downstream packages that us python-ntlm3